_posts/2021-05-10-helm-charts-for-wildfly.adoc

```diff
@@ -0,0 +1,226 @@
+----
+
+After the application is built, we can watch its deployment by running:
+
+[source]
+----
+$ oc get deployment microprofile-config-app -w
+NAME                      READY   UP-TO-DATE   AVAILABLE   AGE
+microprofile-config-app   0/1     1            0           4m4s
+microprofile-config-app   0/1     1            0           4m30s
+microprofile-config-app   1/1     1            1           5m17s
+----
+
+The application is now available and we can query it.
+By default, the Helm Chart will create a `Route` to access the application:
+
+[source]
+----
+$ oc get route
+NAME                      HOST/PORT                                                                        PATH   SERVICES                  PORT    TERMINATION     WILDCARD
+microprofile-config-app   microprofile-config-app-jmesnil1-dev.apps.sandbox-m2.ll9k.p1.openshiftapps.com          microprofile-config-app   <all>   edge/Redirect   None
+----
+
+[NOTE]
+====
+The `HOST/PORT` value is specific to my cluster and your route will have a different value
+====
+
+We can then query the HTTP endpoint by using this information:
+
+[source]
+----
+$ curl -L  http://$(oc get route microprofile-config-app -o jsonpath="{.spec.host}")/config/value
+Hello from OpenShift
+----
+
+The application is up and running.
+Let's now look at the configuration file that the Chart used to install the application.
+
+## Looking at the Helm Release File
+
+Let's deep dive in the https://raw.githubusercontent.com/wildfly/wildfly-charts/main/examples/microprofile-config/microprofile-config-app.yaml[YAML File] that we use to install the application (https://raw.githubusercontent.com/wildfly/wildfly-charts/main/examples/microprofile-config/microprofile-config-app.yaml):
+
+[source,yaml]
+----
+build:
+  uri: https://github.com/wildfly/quickstart.git
+  ref: 23.0.0.Final
+  mode: bootable-jar
+  env:
+  - name: ARTIFACT_DIR
+    value: microprofile-config/target
+  - name: MAVEN_ARGS_APPEND
+    # Use the bootable-jar-openshift profile to ensure that the application
+    # can be deployed on OpenShift but disable JKube as the image will be 
+    # built and deployed by this chart.
+    value: -am -pl microprofile-config -Pbootable-jar-openshift -Djkube.skip=true
+  - name: MAVEN_OPTS
+    value: '-XX:MetaspaceSize=251m -XX:MaxMetaspaceSize=256m'
+deploy:
+  replicas: 1
+  env:
+  - name: CONFIG_PROP
+    value: Hello from OpenShift
+----
+
+The first thing to notice is that the configuration file contains two main sections `build` and `deploy`.
+
+The `build` section focuses on building the application image using an OpenShift `BuildConfig` resource.
+The `BuildConfig` resource pulls the application from a Git repository (based on the `uri` and `ref` fields).
+
+The `mode` field specified which types of application images to build. There are two valid modes: `s2i` (that uses WildFLy S2I Builder and Runtime images) and `bootable-jar`.
```

Comment:
WildFLy ==> WildFly

Reply:
s/The `mode` field specified/The `mode` field specifies